File: pc-parts-admin/src/types/product.ts
export interface Product {
  id: number
  spuCode: string
  name: string
  categoryId: number
  categoryName: string | null
  brandId: number
  brandName: string | null
  description: string | null
  status: number
  createdAt: string
  updatedAt: string
}

export interface ProductCreateRequest {
  spuCode: string
  name: string
  categoryId: number
  brandId: number
  description?: string
  status: number
}

export interface ProductUpdateRequest {
  spuCode: string
  name: string
  categoryId: number
  brandId: number
  description?: string
  status: number
}

File: pc-parts-management/src/main/java/com/example/pcparts/entity/Product.java
package com.example.pcparts.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import java.time.LocalDateTime;
import lombok.Data;

@Data
@TableName("product")
public class Product {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String spuCode;

    private String name;

    private Long categoryId;

    private Long brandId;

    private String description;

    private Integer status;

    @TableLogic
    private Integer deleted;

    private LocalDateTime createdAt;

    private Long createdBy;

    private LocalDateTime updatedAt;

    private Long updatedBy;
}
